modules/tensor_mechanics/src/materials/RankTwoCylindricalComponent.C

```diff
@@ -0,0 +1,60 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "RankTwoCylindricalComponent.h"
+#include "RankTwoScalarTools.h"
+
+registerMooseObject("TensorMechanicsApp", RankTwoCylindricalComponent);
+
+InputParameters
+RankTwoCylindricalComponent::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription("Compute a cylindrical stress/strain property of a RankTwoTensor");
+  params.addRequiredParam<MaterialPropertyName>("rank_two_tensor",
+                                                "The rank two material tensor name");
+  params.addParam<std::string>("calculation_name",
+                               "Optional parameter that allows the user to define "
+                               "multiple mechanics material systems on the same "
+                               "block, i.e. for multiple phases");
+  params.addParam<MooseEnum>(
+      "scalar_type", RankTwoScalarTools::cylindricalOptions(), "Type of scalar output");
+  params.addParam<Point>(
+      "point1", "Start point for axis used to calculate some Direction material tensor quantities");
+  params.addParam<Point>("point2",
+                         "End point for axis used to calculate some material tensor quantities");
+  return params;
+}
+
+RankTwoCylindricalComponent::RankTwoCylindricalComponent(const InputParameters & parameters)
+  : Material(parameters),
+    _tensor(getMaterialProperty<RankTwoTensor>("rank_two_tensor")),
+    _calculation_name(isParamValid("calculation_name") ? getParam<std::string>("calculation_name")
+                                                       : ""),
+    _calculation(declareProperty<Real>(_calculation_name)),
+    _scalar_type(getParam<MooseEnum>("scalar_type")),
+    _point1(isParamValid("point1") ? getParam<Point>("point1") : Point(0, 0, 0)),
+    _point2(isParamValid("point2") ? getParam<Point>("point2") : Point(0, 1, 0)),
+    _direction(Point(0, 0, 0))
```

Comment:
You shouldn't need a `_direction` variable.

Reply:
Based on our discussion, I guess you need this. I think that it's a little confusing for it to be a member variable because it makes it look like it's important. I think I would just create a dummy Point object just before the call to `getCylindricalComponent()`. You don't need to initialize it or anything, just do something like:
```
Point dummy_direction;